kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/ResourcesAdapter.kt

```diff
@@ -36,24 +38,42 @@ internal class ResourcesAdapter() : ListAdapter<Resource, ResourcesAdapter.ViewH
         }
     }
 
+    private fun onSwitchToggled(resource: ViewResource) {
+        activity.onViewResourceToggled(resource)
+    }
+
     class ViewHolder(private val binding: ListItemResourceBinding) : RecyclerView.ViewHolder(binding.root) {
-        fun bind(resource: Resource) {
+        fun bind(
+            resource: ViewResource,
+            onSwitchToggled: (ViewResource) -> Unit,
+        ) {
             binding.resourceNameText.text = resource.name
             binding.addressText.text = resource.address
+            // Without this the item gets reset when out of view, isn't android wonderful?
```

Comment:
Must be because the items get recycled in the view?

Reply:
Yeah it's because recycler views are memory efficient and free state of items that go out of view.

I think the `onBindViewHolder` callback is designed specifically for this purpose, but you may want to double-check:

https://developer.android.com/reference/androidx/recyclerview/widget/RecyclerView.Adapter#onBindViewHolder(VH,int,java.util.List%3Cjava.lang.Object%3E)